public/app/features/alerting/unified/hooks/useCombinedRuleNamespaces.ts

```diff
@@ -67,9 +71,13 @@ export function useCombinedRuleNamespaces(rulesSourceName?: string): CombinedRul
     return rulesSources
       .map((rulesSource): CombinedRuleNamespace[] => {
         const rulesSourceName = isCloudRulesSource(rulesSource) ? rulesSource.name : rulesSource;
-        const promRules = promRulesResponses[rulesSourceName]?.result;
+        let promRules = promRulesResponses[rulesSourceName]?.result;
         const rulerRules = rulerRulesResponses[rulesSourceName]?.result;
 
+        if (rulesSourceName === GRAFANA_RULES_SOURCE_NAME && promRuleNamespaces) {
```

Comment:
```suggestion
        if ( promRuleNamespaces) {
```
So, if `useCombinedRuleNamespaces` we will not use the Redux store and we will use the  `promRuleNamespaces` parameter.

Reply:
Actually, maybe better:
const promRules = promRuleNamespaces??promRulesResponses[rulesSourceName]?.result :